test/common-tests/request-field-validation-api-tests/boolean-field-validation-api-tests.ts

```diff
@@ -24,6 +24,7 @@ export function withBooleanFieldValidationApiTests<RequestBodyItem, RequestBodyI
     });
 
     if (required) {
+      const expectedRequiredFieldError = `${fieldName} should not be empty`;
       it(`returns a 400 response if ${fieldName} is not present`, async () => {
         const [{ [fieldNameSymbol]: _removed, ...requestWithoutTheField }] = validRequestBody;
 
```

Comment:
I've just noticed that these boolean validation tests would fail in the case when `required` is false (we haven't caught this because we only use one boolean value in the API and that's `required`). These tests always check that it `returns a 400 response if ${fieldName} is null`, regardless of the value of `required`, but in reality, the behaviour is the same as for string fields (i.e., if required, we should get a `400` for `undefined` AND `null`, if not, we should get a `2xx` for both). 

The tests should be updated to reflect this, and ideally the `validated-boolean-api-property.decorator.ts` file should be updated to use the new `parseRequiredAndNullable` logic.

Reply:
> we only use one boolean value in the API and that's required

In which case I think the actual fix I'll make is to make sure only ever test that case :) I don't want to assume what the logic should be for non-required boolean fields until we have one